### PR TITLE
Add support for ADO AdvSec validation rules

### PR DESCRIPTION
**Pull request by EasyRhinoMSFT**
This change adds a new filter checkbox for ADO rules. It also changes the behavior of the checkboxes by triggering reprocessing instead of only re-filtering the current results.

I will leave this PR in draft state until I have completed testing.

### CODE REVIEW

**Comment by EasyRhinoMSFT** [comment]
Hi folks, gentle ping...

File: src/SarifWeb/Controllers/ValidationController.cs [hunk 1]
```diff
@@ -72,25 +73,39 @@ public ValidationResponse Post([FromBody] ValidationRequest validationRequest)
         [HttpPost("ValidateFiles")]
         [DisableRequestSizeLimit]
         [RequestFormLimits(MultipartBodyLengthLimit = int.MaxValue, ValueLengthLimit = int.MaxValue)]
-        public IActionResult ValidateFiles(IEnumerable<IFormFile> postedFiles)
+        public IActionResult ValidateFiles(IEnumerable<IFormFile> postedFiles, string kinds) //List<RuleKind> ruleKinds)
```
Comment (shaopeng-gh) [inline src/SarifWeb/Controllers/ValidationController.cs:76]: is this a not needed comment

File: src/SarifWeb/Services/ValidationService.cs [hunk 2]
```diff
@@ -51,15 +54,14 @@ public ValidationResponse Validate(ValidationRequest validationRequest)
                 var validateOptions = new ValidateOptions
                 {
                     OutputFilePath = outputFilePath,
-                    Force = true,
```
Comment (shaopeng-gh) [inline src/SarifWeb/Services/ValidationService.cs:54]: >Force [](http://example.com/codeflow?start=20&length=5)

why remove these 2?

**Review by shaopeng-gh — APPROVED** [review]
:shipit:

File: src/SarifWeb/Services/ValidationService.cs (hunk 2 above)
Comment (EasyRhinoMSFT) [reply src/SarifWeb/Services/ValidationService.cs:54]: No longer present in the SDK class

**Comment by michaelcfanning** [comment]
Please remove Eddy and Mary from owners.txt and add Ross. Shaopeng or Ross have approval perms for this PR, proceed as you see fit.

**Pull request merged**